samples/python/agents/semantickernel/agent.py

```diff
@@ -178,74 +206,55 @@ async def invoke(self, user_input: str, session_id: str) -> dict[str, Any]:
         return self._get_agent_response(response.content)
 
     async def stream(
-        self,
-        user_input: str,
-        session_id: str,
+        self, user_input: str, session_id: str
     ) -> AsyncIterable[dict[str, Any]]:
-        """For streaming tasks we yield the SK agent's invoke_stream progress.
+        """For streaming tasks (like tasks/sendSubscribe), we yield partial progress using SK agent's invoke_stream.
 
         Args:
             user_input (str): User input message.
             session_id (str): Unique identifier for the session.
 
         Yields:
-            dict: A dictionary containing the content, task completion status,
-            and user input requirement.
+            dict: A dictionary containing the content, task completion status, and user input requirement.
         """
         await self._ensure_thread_exists(session_id)
 
-        plugin_notice_seen = False
-        plugin_event = asyncio.Event()
-
-        text_notice_seen = False
         chunks: list[StreamingChatMessageContent] = []
 
-        async def _handle_intermediate_message(
-            message: 'ChatMessageContent',
-        ) -> None:
-            """Handle intermediate messages from the agent."""
-            nonlocal plugin_notice_seen
-            if not plugin_notice_seen:
-                plugin_notice_seen = True
-                plugin_event.set()
-            # An example of handling intermediate messages during function calling
-            for item in message.items or []:
-                if isinstance(item, FunctionResultContent):
-                    print(
-                        f'SK Function Result:> {item.result} for function: {item.name}'
-                    )
-                elif isinstance(item, FunctionCallContent):
-                    print(
-                        f'SK Function Call:> {item.name} with arguments: {item.arguments}'
-                    )
-                else:
-                    print(f'SK Message:> {item}')
-
-        async for chunk in self.agent.invoke_stream(
+        # For the sample, to avoid too many messages, only show one "in-progress" message for each task
+        tool_call_in_progress = False
+        message_in_progress = False
+        async for response_chunk in self.agent.invoke_stream(
             messages=user_input,
             thread=self.thread,
-            on_intermediate_message=_handle_intermediate_message,
         ):
-            if plugin_event.is_set():
-                yield {
-                    'is_task_complete': False,
-                    'require_user_input': False,
-                    'content': 'Processing function calls...',
-                }
-                plugin_event.clear()
-
-            if any(isinstance(i, StreamingTextContent) for i in chunk.items):
-                if not text_notice_seen:
+            if any(
```

Comment:
Please have a look at what is in main. This is no longer handled this way (as of SK >=1.30.0) - `FunctionCallContent` and `FunctionResultContent` are handled via the `on_intermediate_message` callback.

Reply:
merged from the latest main code.